### PR TITLE
refactor: decouple policy pkg from starboard apis

### PR DESCRIPTION
**Pull request by danielpacak**
This will allow us to use the aquasecurity/defsec module
to load and evaluate Rego policies.

Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: pkg/policy/policy.go [hunk 1]
```diff
@@ -73,23 +73,31 @@ func NewMetadata(values map[string]interface{}) (Metadata, error) {
...
-func NewResult(values map[string]interface{}) (Result, error) {
+type Results []Result
+
+// NewMessage constructs new message string based on raw values.
+func NewMessage(values map[string]interface{}) (string, error) {
 	if values == nil {
-		return Result{}, errors.New("values must not be nil")
+		return "", errors.New("values must not be nil")
 	}
 	message, err := requiredStringValue(values, "msg")
```
Comment (chen-keinan) [inline pkg/policy/policy.go:96]: maybe make "msg" a constant with description comment just to make it clear for new readers

File: pkg/policy/policy.go [hunk 2]
```diff
@@ -259,13 +267,13 @@ func (p *Policies) Eval(ctx context.Context, resource client.Object) ([]v1alpha1
 			return nil, fmt.Errorf("failed evaluating Rego deny rule: %s: %w", denyQuery, err)
 		}
 
-		denyResults, hasDenyResults := hasBindings(deny, "res")
-		if hasDenyResults {
-			denyChecks, err := valuesToChecks(md, denyResults)
+		denyValues, hasDenyValues := hasBindings(deny, "res")
```
Comment (chen-keinan) [inline pkg/policy/policy.go:270]: maybe make "res" a constant with description comment just to make it purpose clear for new readers

**Comment by danielpacak** [comment]
It seems that we have a flaky test. Can you check why it's failing @chen-keinan ?

```
--- FAIL: TestControlChecksByScannerChecks (0.00s)
[95](https://github.com/aquasecurity/starboard/runs/5477898995?check_suite_focus=true#step:5:95)
    --- FAIL: TestControlChecksByScannerChecks/_control_checks_by_scanner_checks (0.00s)
```

File: pkg/policy/policy.go (hunk 1 above)
Comment (danielpacak) [reply pkg/policy/policy.go]: Okey, I used constant for  the msg variable.

File: pkg/policy/policy.go (hunk 2 above)
Comment (danielpacak) [reply pkg/policy/policy.go]: Done

**Comment by chen-keinan** [comment]
I have pushed the fix for flaky test in #1009

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**